### PR TITLE
plugins: add new types and client methods

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tibor Vass <teabee89@gmail.com>

Needed by docker/docker#23446

### CODE REVIEW

File: client/plugin.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		}
+
+		fmt.Fprint(out, "Do you grant the above permissions? [y/N] ")
+		reader := bufio.NewReader(in)
+		line, _, err := reader.ReadLine()
+		if err != nil {
+			return err
+		}
+		if strings.ToLower(string(line)) != "y" {
+			return pluginPermissionDenied{name}
```
Comment (crosbymichael) [inline client/plugin.go]: should you delete the plugin off the machine if here?

**Comment by vieux** [comment]
LGTM

**Comment by tiborvass** [comment]
@crosbymichael fixed

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**